Cookbook/Proposals/MultilineArgumentsBracket_SwiftLint_rule.md

```diff
@@ -0,0 +1,121 @@
+
+However, the given set of rules goes a step further. `multiline_arguments_brackets` rule makes it possible achieve what we want, which is, having a trailing bracket at a function call in the new line `)`. However, this rule **always** requires to put trailing bracket `)` in a newline if the function takes more than just one line to be invoked. Even if it's only one argument.
+
+Few example what pieces of the code trigger the rule and how can this be fixed:
+
+ 1. **Trailing closure argument**:
+```swift
+AppsFlyerTracker.shared()?.continue(userActivity, restorationHandler: { restoring in
+        restorationHandler(restoring?.compactMap({ $0 as? UIUserActivityRestoring }))
+})
+```
+The fix:
+```swift
+AppsFlyerTracker.shared()?.continue(
+    userActivity,
+    restorationHandler: { restoring in
+        restorationHandler(restoring?.compactMap({ $0 as? UIUserActivityRestoring }))
+    }
+)
+```
+
+ 2. **Inlined array argument**:
+```swift
+NSLayoutConstraint.activate([
+    imageView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -viewModel.viewSpec.secondaryLogoDistanceFromBottom),
+    imageView.centerXAnchor.constraint(equalTo: view.centerXAnchor)
+])
+```
+The fix:
+```swift
+NSLayoutConstraint.activate(
+    [
+        imageView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -viewModel.viewSpec.secondaryLogoDistanceFromBottom),
+        imageView.centerXAnchor.constraint(equalTo: view.centerXAnchor)
+    ]
+)
+```
+or:
+```swift
+NSLayoutConstraint.activate(
+    [imageView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -viewModel.viewSpec.secondaryLogoDistanceFromBottom),
+     imageView.centerXAnchor.constraint(equalTo: view.centerXAnchor)]
+)
+```
+
+ 3. **Multiline inner argument**:
+```swift
+return accessible.access(InsuranceDetailsService.saveInsuranceDetails(
+    membershipId: membershipId,
+    selectedProvider: selectedProvider,
+    patientId: patientId
+))
+```
+The fix:
+```swift
+return accessible.access(
+    InsuranceDetailsService.saveInsuranceDetails(
+        membershipId: membershipId,
+        selectedProvider: selectedProvider,
+        patientId: patientId
+    )
+)
+```
+
+More examples of changes that would need to be added can be found in [the sample PR](https://github.com/Babylonpartners/babylon-ios/pull/7246/files) I've created against develop. It's not ready to be merged. It's a result of running [SwiftFormat](https://github.com/nicklockwood/SwiftFormat) & couple of manual changes.
+ 
+## Alternatives considered
+
+There are few alternatives:
+
+- drop the one of those four rules (the `multiline_arguments_brackets`). However dropping this rule will allow to have such a function call, which is against our style guide:
```

Comment:
Looking at examples of suggested rules the only ensure that there is no mix on parameters defined on separate lines or in one line, but they still allow first parameter to be defined on the line of function call or declaration:

```swift
func some(paramA: A,
          paramB: B)
```

Personally I'd prefer us to stick with one variant, which is

```swift
func some(
    paramA: A,
    paramB: B
)
```